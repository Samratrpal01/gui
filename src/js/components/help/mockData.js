export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
