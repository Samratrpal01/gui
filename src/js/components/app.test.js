import React from 'react';
import 'jsdom-worker';
import { act } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import App from './app';
import { defaultState, undefineds } from '../../../tests/mockData';
import { render } from '../../../tests/setupTests';

const mockStore = configureStore([thunk]);

describe('App Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      ...defaultState,
      app: {
        ...defaultState.app,
        trackerCode: 'testtracker'
      },
      deployments: {
        ...defaultState.deployments,
        byId: {},
        byStatus: {
          ...defaultState.deployments.byStatus,
          inprogress: {
            ...defaultState.deployments.byStatus.inprogress,
            total: 0
          }
        },
        deploymentDeviceLimit: null
      },
      users: {
        ...defaultState.users,
        currentUser: null
      }
    });
  });

  it('renders correctly', async () => {
    Object.defineProperty(window.document, 'cookie', {
      writable: true,
      value: 'JWT=omnomnom'
    });
    const { baseElement } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as intended', async () => {
    Object.defineProperty(window.document, 'cookie', {
      writable: true,
      value: 'JWT=omnomnom'
    });
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    await act(async () => jest.advanceTimersByTime(900500));
  });
});
