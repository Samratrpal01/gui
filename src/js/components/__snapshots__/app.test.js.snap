// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) and (orientation: landscape) {
  .emotion-0 {
    min-height: 48px;
  }
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 64px;
  }
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
}

.emotion-1 iconButton {
  margin-right: 8px;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #337a87;
  border-radius: 2px;
  font-size: 14px;
  padding: 10px 15px;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(51, 122, 135, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-3:hover {
  colors: #337a87;
}

.emotion-3.MuiButton-text {
  color: rgba(10, 10, 11, 0.78);
}

.emotion-4 {
  display: inherit;
  margin-right: 8px;
  margin-left: -4px;
}

.emotion-4>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-6 {
  display: inherit;
  margin-right: -4px;
  margin-left: 8px;
}

.emotion-6>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  background-color: #f7f7f7;
  border-right: 1px solid #e6f2f1;
}

.emotion-10 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 11px;
  padding-bottom: 11px;
}

.emotion-11.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-11.Mui-selected {
  background-color: rgba(51, 122, 135, 0.08);
}

.emotion-11.Mui-selected.Mui-focusVisible {
  background-color: rgba(51, 122, 135, 0.2);
}

.emotion-11.Mui-disabled {
  opacity: 0.38;
}

.emotion-11.Mui-disabled {
  opacity: 1;
}

.emotion-11.active {
  background-color: #fff;
}

.emotion-11.leftNav.active {
  border-top: 1px solid #d4e9e7;
  border-bottom: 1px solid #d4e9e7;
}

.emotion-11.navLink,
.emotion-11.navLink .MuiListItemText-root {
  color: #969696;
}

.emotion-12 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
  margin-top: 0;
  margin-bottom: 0;
  color: rgba(10, 10, 11, 0.78);
}

.emotion-13 {
  margin: 0;
  line-height: 1.5;
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.9286rem;
  display: block;
}

.emotion-23 {
  padding: 0;
  position: absolute;
  bottom: 30px;
  left: 0;
  right: 0;
}

.emotion-25 {
  padding: 16px 16px 16px 42px;
}

.emotion-31 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 6px;
  margin-bottom: 6px;
  margin-top: 0;
  margin-bottom: 0;
  color: rgba(10, 10, 11, 0.78);
}

.emotion-33 {
  margin: 0;
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.43;
  color: rgba(0, 0, 0, 0.54);
  display: block;
}

.emotion-34 {
  font-size: 13px;
  position: relative;
  top: 6px;
  color: #337a87;
}

<div>
  <div
    id="app"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header emotion-0"
      id="fixedHeader"
      style="min-height: unset; padding-left: 32px; padding-right: 40px;"
    >
      <div
        class="flexbox"
      >
        <a
          href="/"
        >
          <img
            id="logo"
            src="headerlogo.png"
          />
        </a>
        <div
          style="flex-grow: 1;"
        />
        <div
          class="header-section"
          data-mui-internal-clone-element="true"
        >
          <a
            class=""
            href="/devices"
          >
            <span>
              2
            </span>
            <span
              id="limit"
            >
              /
              500
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-testid="DeveloperBoardIcon"
              focusable="false"
              style="margin: 0px 7px 0px 10px; font-size: 20px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9V7h-2V5c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-2h2v-2h-2v-2h2v-2h-2V9h2zm-4 10H4V5h14v14zM6 13h5v4H6zm6-6h4v3h-4zM6 7h5v5H6zm6 4h4v6h-4z"
              />
            </svg>
          </a>
          <a
            href="/devices/pending"
            style="margin-left: 7px;"
          >
            1
             pending
          </a>
        </div>
        <a
          class="header-section"
          href="/deployments/"
        >
          <span>
            0
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium flip-horizontal emotion-1"
            data-testid="RefreshIcon"
            focusable="false"
            style="color: rgb(199, 199, 199); margin: 0px 7px 0px 10px; top: 5px; font-size: 20px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </a>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root header-dropdown emotion-3"
          style="font-size: 14px; margin-left: 4px; text-transform: none;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-4"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-testid="AccountCircleIcon"
              focusable="false"
              style="color: rgb(199, 199, 199);"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </span>
          
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium emotion-6"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m7 10 5 5 5-5z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root emotion-8"
          />
        </button>
      </div>
    </div>
    <div
      class="leftFixed leftNav emotion-9"
    >
      <ul
        class="MuiList-root MuiList-padding emotion-10"
        style="padding: 0px;"
      >
        <a
          aria-current="page"
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-11 active"
          href="/"
          style="padding: 22px 16px 22px 42px;"
        >
          <div
            class="MuiListItemText-root emotion-12"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              Dashboard
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-11"
          href="/devices"
          style="padding: 22px 16px 22px 42px;"
        >
          <div
            class="MuiListItemText-root emotion-12"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              Devices
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-11"
          href="/releases"
          style="padding: 22px 16px 22px 42px;"
        >
          <div
            class="MuiListItemText-root emotion-12"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              Releases
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-11"
          href="/deployments"
          style="padding: 22px 16px 22px 42px;"
        >
          <div
            class="MuiListItemText-root emotion-12"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              Deployments
            </span>
          </div>
        </a>
      </ul>
      <ul
        class="MuiList-root MuiList-padding emotion-23 emotion-10"
      >
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-25 emotion-11"
          href="/help"
        >
          <div
            class="MuiListItemText-root emotion-12"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              Help
            </span>
          </div>
        </a>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-25 emotion-11"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline emotion-31"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-13"
            >
              <div
                class="clickable slightly-smaller"
                data-mui-internal-clone-element="true"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-33"
            >
              <a
                class="emotion-34"
                href="https://docs.mender.io/development/release-information/open-source-licenses"
                rel="noopener noreferrer"
                target="_blank"
              >
                License information
              </a>
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="rightFluid container"
    >
      <div
        class="flexbox centered"
        style="height: 75%;"
      >
        <div
          class="loaderContainer"
        >
          <div
            class="  loader"
          >
            <span
              class="dot dot_1"
            />
            <span
              class="dot dot_2"
            />
            <span
              class="dot dot_3"
            />
            <span
              class="dot dot_4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
